modules/icon/icon.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+// This file was ported from https://github.com/Claudiohbsantos/github-material-icons-extension/blob/ff97e50980/src/lib/replace-icon.js
+// to go with small changes.
+
+package icon
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/json"
+)
+
+type Map struct {
+	FileNames      map[string]string `json:"fileNames"`
+	FolderNames    map[string]string `json:"folderNames"`
+	FileExtensions map[string]string `json:"fileExtensions"`
+	LanguageIds    map[string]string `json:"languageIds"`
+	Light          Light             `json:"light"`
+}
+
+type Light struct {
+	FileNames      map[string]string `json:"fileNames"`
+	FolderNames    map[string]string `json:"folderNames"`
+	FileExtensions map[string]string `json:"fileExtensions"`
+}
```

Comment:
This type seems to be unused:
You deserialize it, and then you do nothing with it…

Reply:
Not even speaking about these other things I see in the json file that seem to be missing completely, i.e. high contrast.

Reply:
Thanks, I'll drop what I'm not using. Yes, the file has more features including icons for light themes. In this PR I will focus only on adding the default set of icons, a following PR could make use of the full capabilities of this icon set.